packages/node/src/utils.ts

```diff
@@ -35,3 +38,51 @@ export async function getPeersAddressFromAppInstanceID(
   const owners = stateChannel.multisigOwners;
   return owners.filter(owner => owner !== selfAddress);
 }
+
+export async function confirmAppInstanceExists(
+  store: Store,
+  appInstanceId: string
+): Promise<boolean> {
+  const stateChannel = await store.getChannelFromAppInstanceID(appInstanceId);
+  if (!stateChannel) {
+    console.log("nope");
+    return Promise.reject(ERRORS.NO_APP_INSTANCE_FOR_APP_INSTANCE_ID);
+  }
+  return true;
+}
+
+export function getAppContractToApplyAction(
+  appInstance: AppInstance,
+  provider: BaseProvider
+): Contract {
+  if (
+    !appInstance.appInterface.addr ||
+    appInstance.appInterface.addr.trim() === ""
+  ) {
+    throw Error(ERRORS.UNSPECIFIED_CONTRACT_ADDRESS);
+  }
+
+  if (
+    !appInstance.appInterface.stateEncoding ||
+    appInstance.appInterface.stateEncoding.trim() === ""
+  ) {
+    throw Error(ERRORS.UNSPECIFIED_CONTRACT_ADDRESS);
+  }
+
+  const abi = [
+    `function applyAction(${appInstance.appInterface.stateEncoding} state, ${
+      appInstance.appInterface.actionEncoding
+    } action) public pure returns (bytes)`
+  ];
+
+  console.log("abi for new contract");
+  console.log(abi);
+  return new Contract(appInstance.appInterface.addr, abi, provider);
+}
+
+export function decodeAppState(
+  encodedAppState: string,
+  appStateEncoding: string
+): AppState {
+  return defaultAbiCoder.decode([appStateEncoding], encodedAppState)[0];
+}
```

Comment:
This should probably be a method on `AppInstance`?

Reply:
Yup, will update